src/zarr/core/array.py

```diff
@@ -1556,6 +1608,19 @@ async def setitem(
         )
         return await self._set_selection(indexer, value, prototype=prototype)
 
+    @property
+    def oindex(self) -> AsyncOIndex[T_ArrayMetadata]:
```

Comment:
I chose this API to try to follow this pattern:

- `Array.__getitem__` (exists)
- `Array.oindex.__getitem__` (exists)
- `Array.vindex.__getitem__` (exists)
- `AsyncArray.getitem` (exists)
- `AsyncArray.oindex.getitem` (new)
- `AsyncArray.vindex.getitem` (new)

because python doesn't let you make an async version of the `__getitem__` magic method.

Reply:
Can we update the sync versions to use `sync(self._async_array.oindex.getitem)` instead of going to `_get_selection` directly? That will get us some test coverage